nsdb-cluster/src/main/scala/io/radicalbit/nsdb/cluster/NSDBAkkaCluster.scala

```diff
@@ -49,25 +52,39 @@ trait NSDBAActors { this: NSDBAkkaCluster =>
 
   implicit val executionContext: ExecutionContextExecutor = system.dispatcher
 
-  private val databaseActorGuardian = system.actorOf(
-    Props(classOf[DatabaseActorsGuardian]),
+  system.actorOf(
+    ClusterSingletonManager.props(singletonProps = Props(classOf[DatabaseActorsGuardian]),
+                                  terminationMessage = PoisonPill,
+                                  settings = ClusterSingletonManagerSettings(system)),
     name = "databaseActorGuardian"
   )
 
-  Future
-    .sequence(
-      Seq((databaseActorGuardian ? GetMetadataCache).mapTo[ActorRef],
-          (databaseActorGuardian ? GetSchemaCache).mapTo[ActorRef]))
-    .onComplete {
-      case Success(metadataCache :: schemaCache :: Nil) =>
-        system.actorOf(
-          ClusterListener.props(NodeActorsGuardian.props(metadataCache, schemaCache)),
-          name = s"cluster-listener_${createNodeName(Cluster(system).selfMember)}"
-        )
-      case _ =>
-        system.log.error("Error retrieving caches, terminating system.")
-        system.terminate()
-    }
+  DistributedData(system).replicator
+
+  system.actorOf(
+    ClusterSingletonProxy.props(singletonManagerPath = "/user/databaseActorGuardian",
+                                settings = ClusterSingletonProxySettings(system)),
+    name = "databaseActorGuardianProxy"
+  )
+
+  lazy val nodeName: String = createNodeName(Cluster(system).selfMember)
+
+  (for {
+    databaseActorGuardian <- system.actorSelection("user/databaseActorGuardianProxy").resolveOne()
+    metadataCache         <- (databaseActorGuardian ? GetMetadataCache(nodeName)).mapTo[ActorRef]
+    schemaCache           <- (databaseActorGuardian ? GetSchemaCache(nodeName)).mapTo[ActorRef]
+  } yield {
+    system.log.debug("caches got")
```

Comment:
The log seems incomplete (?)

Reply:
It was a log I used to figure out an issue in tests. I can definitely improve the message. Thanks